environment/os/install_deps.sh

```diff
@@ -0,0 +1,140 @@
+    local distro_script=$1
+    echo "Running script for $distro_script"
+    if [ "$toolchain_deps_flag" = true ]; then
+        sudo ./environment/os/$distro_script check TOOLCHAIN_RUN_DEPS
+        sudo ./environment/os/$distro_script install TOOLCHAIN_RUN_DEPS
+    fi
+    if [ "$build_deps_flag" = true ]; then
+        sudo ./environment/os/$distro_script install MEMGRAPH_BUILD_DEPS
+    fi
+}
+
+# Detect OS, version, and architecture
+if [ -f /etc/os-release ]; then
+    . /etc/os-release
+    OS=$ID
+    VER=$VERSION_ID
+elif type lsb_release >/dev/null 2>&1; then
+    OS=$(lsb_release -si)
+    VER=$(lsb_release -sr)
+elif [ -f /etc/lsb-release ]; then
+    . /etc/lsb-release
+    OS=$DISTRIB_ID
+    VER=$DISTRIB_RELEASE
+elif [ -f /etc/debian_version ]; then
+    OS=Debian
+    VER=$(cat /etc/debian_version)
+else
+    echo "OS not identified"
+    exit 1
+fi
+
+ARCH=$(uname -m)
+case $ARCH in
+    arm*|aarch64)
+        ARCH="arm"
+        ;;
+    *)
+        ARCH=""
+        ;;
+esac
+
+# Normalize OS name to lowercase
+OS=$(echo "$OS" | tr '[:upper:]' '[:lower:]')
+
+# Map OS, version, and architecture to script name
+case "$OS" in
+    debian)
+        case "$VER" in
+            10) run_script "debian-10.sh" ;;
+            11) 
+                if [ "$ARCH" = "arm" ]; then
+                    run_script "debian-11-arm.sh"
+                else
+                    run_script "debian-11.sh"
+                fi
+                ;;
+            12) 
+                if [ "$ARCH" = "arm" ]; then
+                    run_script "debian-12-arm.sh"
+                else
+                    run_script "debian-12.sh"
+                fi
+                ;;
+            *) echo "Unsupported Debian version: $VER"; exit 1 ;;
+        esac
+        ;;
+    ubuntu)
+        case "$VER" in
+            18.04) run_script "ubuntu-18.04.sh" ;;
+            20.04) run_script "ubuntu-20.04.sh" ;;
+            22.04) 
+                if [ "$ARCH" = "arm" ]; then
+                    run_script "ubuntu-22.04-arm.sh"
+                else
+                    run_script "ubuntu-22.04.sh"
+                fi
+                ;;
+            *) echo "Unsupported Ubuntu version: $VER"; exit 1 ;;
+        esac
+        ;;
+    centos)
+        case "$VER" in
+            7) run_script "centos-7.sh" ;;
+            9) run_script "centos-9.sh" ;;
+            *) echo "Unsupported CentOS version: $VER"; exit 1 ;;
+        esac
+        ;;
+    fedora)
+        case "$VER" in
+            36) run_script "fedora-36.sh" ;;
+            38) run_script "fedora-38.sh" ;;
+            39) run_script "fedora-39.sh" ;;
+            *) echo "Unsupported Fedora version: $VER"; exit 1 ;;
+        esac
+        ;;
+    "amzn")
+        case "$VER" in
+            2) run_script "amzn-2.sh" ;;
+            *) echo "Unsupported Amazon Linux version: $VER"; exit 1 ;;
+        esac
+        ;;
+    *)
+        echo "Unsupported OS: $OS"
+        exit 1
+        ;;
+esac
```

Comment:
Firstly, rocky-9.3 is missing.  Also I would suggest approaching this in a different way. 
Instead of a case for OSs and nested case for versions, you can leverage the way we named the scripts.
Every script can be selected with `"${OS}-${VER}.sh"`, or `"${OS}-${VER}-arm.sh"` for arm machines.
If you add an array SUPPORTED_OS to the top of the script, you can check the provided OS and VER against that array.
I did a similar thing in another script
https://github.com/memgraph/memgraph/blob/master/release/package/mgbuild.sh#L14
https://github.com/memgraph/memgraph/blob/master/release/package/mgbuild.sh#L157

Reply:
I started working on this when SUPPORTED_OS_V4 was the latest, and in the meantime SUPPORTED_OS_V5 became latest, so I missed rocky-9.3 even exist! Will include it + follow your way from mgbuild.sh

Reply:
did exactly this + tested rocky9.3 => much easier :)